### PR TITLE
Cut release v0.45.0

### PR DESCRIPTION
**Pull request by lilic**
I used [this](https://github.com/prometheus-operator/prometheus-operator/compare/release-0.44...master?expand=1) to get the changes. Please double-check in case I missed any relevant, I only included user facing changes not any docs updates.

@prometheus-operator/prometheus-operator-reviewers please have a look, thanks!

### CODE REVIEW

**Comment by lilic** [comment]
I believe after this is merged the branches are created, I have not done this since we started creating those?

File: RELEASE.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Release cadence of first pre-releases being cut is 6 weeks.
 | v0.44   | 2020-12-02                                 | Pawel Krupa (GitHub: @paulfantom)           |
 | v0.45   | 2021-01-13                                 | Lili Cosic (GitHub: @lilic)                 |
 | v0.46   | 2021-02-24                                 | **searching for volunteer**                 |
```
Comment (lilic) [inline RELEASE.md:16]: We are looking for someone to cut the next one 😸

Comment (paulfantom) [reply RELEASE.md:16]: if no one volounteers, I can do it ;)

**Comment by paulfantom** [comment]
> I believe after this is merged the branches are created

Yes, after this is merged you manually create a release branch and push tags as described in https://github.com/prometheus-operator/prometheus-operator/blob/master/RELEASE.md#publish-the-new-release

**Review by paulfantom — APPROVED** [review]
lgtm

File: RELEASE.md (hunk 1 above)
Comment (lilic) [reply RELEASE.md:16]: Will leave it as searching for now and lets see if anyone volunteers. Otherwise you can add yourself next time :)

**Pull request merged**